zha/application/platforms/__init__.py

```diff
@@ -355,6 +355,11 @@ def _init_from_quirks_metadata(self, entity_metadata: EntityMetadata) -> None:
         elif has_command_name:
             self._unique_id_suffix = entity_metadata.command_name
 
+        if entity_metadata.unique_id_suffix:
+            self._unique_id_suffix = (
+                f"{self._unique_id_suffix}-{entity_metadata.unique_id_suffix}"
+            )
```

Comment:
You mean that instead of appending the quirks v2 provided `unique_id_suffix` to the existing `unique_id_suffix` (created in the lines above), that we just replace it?
I guess that could work. The "attribute_name + extra suffix" is likely a bit unnecessary.

Changing that should allow quirks v2 to take over ZHA/"v1" entities.